src/sentry_sync.c

```diff
@@ -200,6 +199,51 @@ sentry__bgworker_submit(sentry_bgworker_t *bgw,
     return 0;
 }
 
+int
+sentry__bgworker_foreach_matching(sentry_bgworker_t *bgw,
+    sentry_task_function_t exec_func,
+    bool (*callback)(void *task_data, void *data), void *data)
+{
+    sentry__mutex_lock(&bgw->task_lock);
```

Comment:
Could this deadlock if the mutex is held at the time of the crash?

Reply:
Hm, now that you are mentioning this. The inproc backend has special code that switches the mutex implementation (and malloc) in case of a signal handler. According to the comment, it’s supposed to avoid deadlocks, and give the signal handler thread access.

I’m a bit surprised that there are no issues with the breakpad backend. Isn’t that running inside the signal handler?

Reply:
Yes, Breakpad registers signal handlers which eventually invoke your callback. I also missed this in the initial review -- we probably have to add this lock override, too.

I was looking for it at the right spot, but didn't realize that it was `sentry__block_for_signal_handler` what I was looking at :)